src/MongoDB/Manager.c

```diff
@@ -200,6 +200,15 @@ static void php_phongo_manager_prep_uri_options(zval *options TSRMLS_DC) /* {{{
 				continue;
 			}
 
+            if (!strcasecmp(ZSTR_VAL(string_key), MONGOC_URI_READPREFERENCE)) {
+				ZVAL_DEREF(option);
+				SEPARATE_ZVAL_NOREF(option);
+				if (Z_TYPE_P(option) != IS_STRING) {
+                    phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Unsupported readPreference type, a string must be given");
```

Comment:
another question about validating bson value's types, I went with the source code but I couldn't found any clue about it except the `php_phongo_bson_state` struct and `php_phongo_bson_to_zval_ex` function then check the type using the ordinary php way which I'm sure that this is not the proper way to check about the bson value